core/inc/gateway.h

```diff
@@ -46,6 +46,15 @@ DEFINE_ENUM(GATEWAY_ADD_LINK_RESULT, GATEWAY_ADD_LINK_RESULT_VALUES);
  */
 DEFINE_ENUM(GATEWAY_START_RESULT, GATEWAY_START_RESULT_VALUES);
 
+#define GATEWAY_UPDATE_FROM_JSON_RESULT_VALUES \
+    GATEWAY_UPDATE_FROM_JSON_SUCCESS, \
+    GATEWAY_UPDATE_FROM_JSON_ERROR
```

Comment:
It would be great to add `GATEWAY_UPDATE_FROM_JSON_INVALID_ARG` and `GATEWAY_UPDATE_FROM_JSON_MEMORY`, so callers can do the right thing depending on the result.